### ISSUE TITLE
Bug in query param collection serialization

### ISSUE THREAD

**Issue by R0nd** [Azure/ms-rest-js#160]
I've encountered a bug leading to only the last item of a collection-typed query parameter being serialized into the query when using collectionFormat:"multi"

The offending code segment I've traced to (serviceClient.ts):
```typescript
if (queryParameter.collectionFormat === QueryCollectionFormat.Multi) {
                if (queryParameterValue.length === 0) {
                  queryParameterValue = "";
                } else {
                  for (const item of queryParameterValue) {
                    queryParameterValue = (item == undefined ? "" : item.toString());
                  }
```

The issue has been encountered in v0.8.* but the code seems to be unchanged from there.

**Comment by R0nd** [comment]
Reproduced the issue in v0.12.293 using a client generated with autorest.
A simple 
```typescript
client.apiValuesGet({ value: [1, 2, 3] })
```
yields querystring ?value=3

**Comment by R0nd** [comment]
The URLQuery class needs to be reworked to support collectionFormat:"multi" as it stores query parameters in a map with the parameter name as a key.

**Comment by RikkiGibson** [comment]
Thanks for bringing this to our attention. Maybe we can allow the value for a query parameter in URLQuery to be an array to handle the "multi" case?

**Comment by R0nd** [comment]
I've created PR #163 with a fix and appropriate test coverage

**Comment by RikkiGibson** [comment]
Excellent, I'll be sure to have a look today.

**Comment by RikkiGibson** [comment]
Fixed in #163
